docs/etiquette/programming-standards.md

```diff
@@ -0,0 +1,135 @@
+---
```

Comment:
@F-said and @jessb0t I think this programming standards doc is a really, really great start and good for pushing to main as is. That said, we will want to refine/expand this doc further with additional details and examples to make it easier to understand by novices. but again, no need to hold up pushing to main as is!